File: app/src/main/java/com/worldventures/dreamtrips/wallet/ui/records/detail/CardDetailsPresenter.java
package com.worldventures.dreamtrips.wallet.ui.records.detail;

import com.worldventures.dreamtrips.wallet.ui.common.base.WalletPresenter;

public interface CardDetailsPresenter extends WalletPresenter<CardDetailsScreen> {

   void goBack();

   void updateNickName();

   void onDeleteCardClick();

   void payThisCard();

   void onChangeDefaultCardConfirmed();

   void onChangeDefaultCardCanceled();

   void onDeleteCardConfirmed();

   void onCardIsReadyDialogShown();

   void validateRecordName(String name);

   void changeDefaultCard(boolean isDefault);
}

File: wallet/src/main/java/com/worldventures/wallet/ui/records/detail/CardDetailsPresenter.kt
package com.worldventures.wallet.ui.records.detail

import com.worldventures.wallet.ui.common.base.WalletPresenter

interface CardDetailsPresenter : WalletPresenter<CardDetailsScreen> {

   fun goBack()

   fun observeRecordChanges(recordId: String)

   fun updateNickname(recordId: String, nickname: String)

   fun fetchDefaultRecord()

   fun onDeleteCardClick()

   fun payThisCard(recordId: String)

   fun onChangeDefaultCardConfirmed(recordId: String)

   fun onChangeDefaultCardCanceled()

   fun onDeleteCardConfirmed(recordId: String)

   fun onCardIsReadyDialogShown()

   fun validateRecordName(name: String)

   fun changeDefaultCard(isDefault: Boolean, recordId: String, defaultRecordDetail: DefaultRecordDetail?)
}
